hadoop-ozone/recon/src/test/java/org/apache/hadoop/ozone/recon/tasks/TestNSSummaryTaskWithOBS.java

```diff
@@ -0,0 +1,548 @@
+  @Nested
+  public class TestReprocess {
+
+    private NSSummary nsSummaryForBucket1;
+    private NSSummary nsSummaryForBucket2;
+
+    @BeforeEach
+    public void setUp() throws IOException {
+      // write a NSSummary prior to reprocess
+      // verify it got cleaned up after.
+      NSSummary staleNSSummary = new NSSummary();
+      RDBBatchOperation rdbBatchOperation = new RDBBatchOperation();
+      reconNamespaceSummaryManager.batchStoreNSSummaries(rdbBatchOperation, -1L,
+          staleNSSummary);
+      reconNamespaceSummaryManager.commitBatchOperation(rdbBatchOperation);
+
+      // Verify commit
+      assertNotNull(reconNamespaceSummaryManager.getNSSummary(-1L));
+
+      // reinit Recon RocksDB's namespace CF.
+      reconNamespaceSummaryManager.clearNSSummaryTable();
+
+      nSSummaryTaskWithOBS.reprocessWithOBS(reconOMMetadataManager);
+      assertNull(reconNamespaceSummaryManager.getNSSummary(-1L));
+
+      nsSummaryForBucket1 =
+          reconNamespaceSummaryManager.getNSSummary(BUCKET_ONE_OBJECT_ID);
+      nsSummaryForBucket2 =
+          reconNamespaceSummaryManager.getNSSummary(BUCKET_TWO_OBJECT_ID);
+      assertNotNull(nsSummaryForBucket1);
+      assertNotNull(nsSummaryForBucket2);
+    }
+
+    @Test
+    public void testReprocessNSSummaryNull() throws IOException {
+      assertNull(reconNamespaceSummaryManager.getNSSummary(-1L));
+    }
+
+    @Test
+    public void testReprocessGetFiles() {
+      assertEquals(3, nsSummaryForBucket1.getNumOfFiles());
+      assertEquals(2, nsSummaryForBucket2.getNumOfFiles());
+
+      assertEquals(KEY_ONE_SIZE + KEY_TWO_OLD_SIZE + KEY_THREE_SIZE,
+          nsSummaryForBucket1.getSizeOfFiles());
+      assertEquals(KEY_FOUR_SIZE + KEY_FIVE_SIZE,
+          nsSummaryForBucket2.getSizeOfFiles());
+    }
+
+    @Test
+    public void testReprocessFileBucketSize() {
+      int[] fileDistBucket1 = nsSummaryForBucket1.getFileSizeBucket();
+      int[] fileDistBucket2 = nsSummaryForBucket2.getFileSizeBucket();
+      assertEquals(ReconConstants.NUM_OF_FILE_SIZE_BINS,
+          fileDistBucket1.length);
+      assertEquals(ReconConstants.NUM_OF_FILE_SIZE_BINS,
+          fileDistBucket2.length);
+
+      // Check for 1's and 0's in fileDistBucket1
+      int[] expectedIndexes1 = {0, 1, 40};
+      for (int index = 0; index < fileDistBucket1.length; index++) {
+        if (contains(expectedIndexes1, index)) {
+          assertEquals(1, fileDistBucket1[index]);
+        } else {
+          assertEquals(0, fileDistBucket1[index]);
+        }
+      }
+
+      // Check for 1's and 0's in fileDistBucket2
+      int[] expectedIndexes2 = {0, 2};
+      for (int index = 0; index < fileDistBucket2.length; index++) {
+        if (contains(expectedIndexes2, index)) {
+          assertEquals(1, fileDistBucket2[index]);
+        } else {
+          assertEquals(0, fileDistBucket2[index]);
+        }
+      }
+    }
+
+  }
+
+  /**
+   * Nested class for testing NSSummaryTaskWithLegacy process.
```

Comment:
```suggestion
   * Nested class for testing NSSummaryTaskWithOBS process.
```

Reply:
Changed thanks

Reply:
I have made the changes for this!
Not sure why its not showing it as outdated.